src/cdsetool/credentials.py

```diff
@@ -22,12 +22,6 @@ class InvalidCredentialsException(Exception):
     """
 
 
-class NoTokenException(Exception):
-    """
-    Raised when no token is available
-    """
-
-
```

Comment:
Could you do something like this, to make for a safer release?

```python

class DeprecatedNoTokenException(Exception):
    """
    deprecated
    """
def NoTokenException(*args, **kwargs):
    from warnings import warn
    warn("Warning! NoTokenException is deprecated, and will be removed in the next major release.")
    return DeprecatedNoTokenException(*args, **kwargs)
```

Reply:
It seems fair to remain somewhat compatible for a while.

I have no idea how your code works under the hood, but I added the lines with a capitalized "D" in the pydoc.

Reply:
Its essentially just a wrapper around the constructor of the old class, which outputs the warning message before running the old class constructor.
Just to help users transition, instead of receiving import errors in their code.

Reply:
Ok, fixed, and passed linting now too.